uplink/ecclient/client.go

```diff
@@ -213,42 +213,35 @@ func (ec *ecClient) putPiece(ctx, parent context.Context, limit *pb.AddressedOrd
 		)
 		return nil, err
 	}
-	defer func() {
-		if ctx.Err() != nil || err != nil {
-			hash = nil
-			err = errs.Combine(err, upload.Cancel(ctx))
-			return
-		}
-		h, closeErr := upload.Commit(ctx)
-		hash = h
-		err = errs.Combine(err, closeErr)
-	}()
 
 	_, err = sync2.Copy(ctx, upload, data)
 	// Canceled context means the piece upload was interrupted by user or due
 	// to slow connection. No error logging for this case.
-	if ctx.Err() == context.Canceled {
-		if parent.Err() == context.Canceled {
-			ec.log.Info("Upload to node canceled by user", zap.String("NodeID", storageNodeID.String()))
+	if err != nil {
+		if err == context.Canceled {
+			if parent.Err() == context.Canceled {
+				ec.log.Info("Upload to node canceled by user", zap.String("NodeID", storageNodeID.String()))
+			} else {
+				ec.log.Debug("Node cut from upload due to slow connection", zap.String("NodeID", storageNodeID.String()))
+			}
 		} else {
-			ec.log.Debug("Node cut from upload due to slow connection", zap.String("NodeID", storageNodeID.String()))
-		}
-		err = context.Canceled
-	} else if err != nil {
-		nodeAddress := "nil"
-		if limit.GetStorageNodeAddress() != nil {
-			nodeAddress = limit.GetStorageNodeAddress().GetAddress()
+			nodeAddress := "nil"
```

Comment:
Maybe use empty string rather than `nil`, sentinel values can easily propagate into the rest of system in weird ways.

Reply:
I don't know why it was set to `"nil"` value, but I mostly left how it was.

I'll do the change